### DISCUSSION TITLE
[bug]: How to let other element follow the animation to take hided place smoothly?

### DISCUSSION THREAD

**Discussion by sxhuan** [pmndrs/react-spring#2169] [Support]
### Which react-spring target are you using?

- [X] `@react-spring/web`
- [ ] `@react-spring/three`
- [ ] `@react-spring/native`
- [ ] `@react-spring/konva`
- [ ] `@react-spring/zdog`

### What version of react-spring are you using?

9.7.2

### What's Wrong?

I am implementing a drawer, which pushes the content, I make the drawer animated, but the content does not follow smoothly like CSS transition, it has to wait until the animation is finished. is there any way to fix this?

https://github.com/pmndrs/react-spring/assets/16263180/3578e8bb-059f-4dc3-8920-d83a9dcbe86a

### To Reproduce

drawer use useSpring, animated with position x, onRest callback to close the drawer

### Expected Behaviour

expect content to follow the animation like CSS transition.

### Link to repo

N/A

**Comment by joshuaellis** [comment]
It'll be an issue in your code, but without seeing it I doubt anyone will be able to offer you advice :)

**Comment by MUK-Dev** [comment]
If you are using `flex` to layout your drawer and the main body component. You should try giving `flex-grow: 1` to your main component.
